lib/panmaster/include/panmaster/pan.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef PAN_H
+#define PAN_H
+
+#include <dpl/dpl.h>
+#include <uwb/uwb.h>
+#include <uwb/uwb_ftypes.h>
+#include "panmaster.h"
+#include "pan_req_resp.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef union{
+//! Structure containing pan response frame format
+    struct _pan_frame_t{
+        //! Structure of IEEE blink frame
+        struct _ieee_blink_frame_t;
+        uint8_t rpt_count:4;                 //!< Repeat level
+        uint8_t rpt_max:4;                   //!< Repeat max level
+        uint16_t code;                       //!< Package type code
+        struct pan_req_resp req;
+    }__attribute__((__packed__, aligned(1)));
+    uint8_t array[sizeof(struct _pan_frame_t)];
+}pan_frame_t;
+
+typedef struct _uwb_pan_config_t{
+    uint32_t rx_holdoff_delay;
+    uint16_t rx_timeout_period;
+    uint32_t tx_holdoff_delay:28;
+    uint32_t role:4;
+    uint16_t lease_time;
+    uint16_t network_role;
+}uwb_pan_config_t;
+
+typedef struct _uwb_pan_control_t{
+    uint16_t postprocess:1;
+    uint16_t request_cb:1;
+}uwb_pan_control_t;
+
+typedef struct _uwb_pan_status_t{
+    uint16_t selfmalloc:1;
+    uint16_t initialized:1;
+    uint16_t valid:1;
+    uint16_t start_tx_error:1;
+    uint16_t lease_expired:1;
+}uwb_pan_status_t;
+
+struct uwb_pan_instance {
+    struct uwb_dev * dev_inst;
+    struct uwb_mac_interface cbs;
+    struct dpl_sem sem;
+    uwb_pan_status_t status;
+    uwb_pan_control_t control;
+    uwb_pan_request_cb_func_t request_cb;
+    struct dpl_event postprocess_event;
+    struct dpl_callout pan_lease_callout_expiry;
+    uwb_pan_config_t * config;
+    uint16_t nframes;
+    uint16_t idx;
+    pan_frame_t * frames[];
+};
+
+#ifdef __cplusplus
+}
+
+#endif
+#endif /* PAN_H */
```

Comment:
It's a bit ugly to have another copy of this and the pan_req_resp header here. I think these data structures may have to go into a separate package that both uwb_pan and panmaster depend upon.